prism_compile.c

```diff
@@ -1852,6 +1857,63 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
                 }
             }
         }
+
+        if (begin_node->rescue_clause) {
+            pm_scope_node_t rescue_scope_node;
+            pm_scope_node_init((pm_node_t *)begin_node->rescue_clause, &rescue_scope_node, scope_node, parser);
+
+            rb_iseq_t *rescue_iseq = NEW_CHILD_ISEQ(rescue_scope_node,
+                                                    rb_str_concat(rb_str_new2("rescue in"),
+                                                                  ISEQ_BODY(iseq)->location.label),
+                                                    ISEQ_TYPE_RESCUE, 1);
+            lstart->rescued = LABEL_RESCUE_BEG;
+            lend->rescued = LABEL_RESCUE_END;
+            ADD_LABEL(ret, lstart);
+
+            bool prev_in_rescue = ISEQ_COMPILE_DATA(iseq)->in_rescue;
+            ISEQ_COMPILE_DATA(iseq)->in_rescue = true;
+            {
+                pm_node_list_t node_list = begin_node->statements->body;
+                if (node_list.size > 0) {
+                    for (size_t index = 0; index < node_list.size - 1; index++) {
+                        PM_COMPILE_POPPED(node_list.nodes[index]);
+                    }
+                    PM_COMPILE_NOT_POPPED(node_list.nodes[node_list.size - 1]);
+                }
+                else {
+                    PM_PUTNIL;
+                }
+            }
+            ISEQ_COMPILE_DATA(iseq)->in_rescue = prev_in_rescue;
+
+            if (begin_node->else_clause) {
+                PM_POP_UNLESS_POPPED;
+                PM_COMPILE((pm_node_t *)begin_node->else_clause);
+            }
+
+            ADD_LABEL(ret, lend);
+            ADD_INSN(ret, &dummy_line_node, nop);
+            ADD_LABEL(ret, lcont);
+
+            if (begin_node->else_clause) {
```

Comment:
If we're calling `PM_POP_IF_POPPED` in either case, do we need this if / else block?